### ISSUE TITLE
'No source root found' error for module

### ISSUE THREAD

**Issue by gtaylor1981** [scoverage/scoverage-maven-plugin#191]
I just tried upgrading from v1.4.11 to the newest v2.0.1 and have encountered a problem.
I recreated this in a minimal project (called `my-project`) containing a subfolder with a single Scala module (`my-module`).
When I run `mvn clean install` it fails with the error:
```
No source root found for 'C:\GIT\my-project\my-module\my-module\src\main\scala\org\example\MyClass.scala'
(source roots: 'C:\GIT\my-project\my-module\src\main\scala\', 'C:\GIT\my-project\my-module\target\generated-sources\annotations\')
```

The module name appears twice in the path (`...\my-module\my-module\...`) which looks like an error to me.

The root POM is almost empty, and the module POM has the plugin configured as follows:
```html
<plugin>
    <groupId>org.scoverage</groupId>
    <artifactId>scoverage-maven-plugin</artifactId>
    <version>2.0.1</version>
    <configuration>
        <additionalForkedProjectProperties>skipTests=false</additionalForkedProjectProperties>
        <highlighting>true</highlighting>
    </configuration>
    <executions>
        <execution>
            <id>scoverage-report</id>
            <goals>
                <goal>report</goal>
            </goals>
            <phase>package</phase>
        </execution>
    </executions>
</plugin>
```

I attached the project and module POM files if it helps:
[project-pom.txt](https://github.com/scoverage/scoverage-maven-plugin/files/13872446/project-pom.txt)
[module-pom.txt](https://github.com/scoverage/scoverage-maven-plugin/files/13872452/module-pom.txt)

I do see a `scoverage-data` folder created in `C:\GIT\my-project\my-module\target\` as expected.
This contains the `source.roots` file which looks like:
```
C:\GIT\my-project\my-module\src\main\java
C:\GIT\my-project\my-module\src\main\scala
C:\GIT\my-project\my-module\target\generated-sources\annotations
```
Also in this folder is the `scoverage.coverage` file which contains the source path:
```
my-module\\src\\main\\scala\\org\\example\\MyClass.scala
```

This looks ok, but perhaps it is being appended to the module path (`C:\GIT\my-project\my-module`) which then causes the duplicated module name in the middle?

Do I need to change something in my setup, or is this a bug in the plugin?
Please let me know if I can provide any further info. And many thanks for resurrecting the plugin and releasing the new version.

**Comment by jozic** [comment]
@gtaylor1981 
from a quick check, seems like if the scoverage plugin is defined in parent's pom it works okay, but if it's defined in module's pom, then things go wrong
I will try to investigate why it happens a bit later, but could you please test it from your side and push plugin declaration and configuration to the parent?

**Comment by gtaylor1981** [comment]
@jozic Hi, thanks for looking into this.
I tried your suggestion of moving the config to the root, but found it made no difference.

To double check this, I also cloned the repo to be able to run the integration tests and I noticed there are no tests covering my situation, i.e. sub-modules in Scala 3.
I found the most similar one to my situation is [`test_aggregate`](https://github.com/scoverage/scoverage-maven-plugin/tree/main/src/it/test_aggregate) so in its main POM, I changed its `properties` to use Scala 3 (in the same way the other Scala 3 tests do, with these three properties):
```html
<scala.compat.version>3</scala.compat.version>
<scala.version>3.3.1</scala.version>
<scala.library.artifact.id>scala3-library_3</scala.library.artifact.id>
```

(I also had to update the `module01` unit test to Scala 3 by changing `def test1() {` to `def test1(): Unit = {`).

Running this integration test reproduced my error message:  `No source root found for 'C:\GIT\scoverage-maven-plugin\target\it\test_aggregate\module01\module01\src\main\scala\pkg01\HelloService1.scala'`

Comparing the `scoverage.coverage` files from this integration test using Scala 2 and Scala 3 there is a difference in the source path:
Scala 2 has `src\main\scala\pkg01\HelloService1.scala`
Scala 3 has `module01\\src\\main\\scala\\pkg01\\HelloService1.scala`

---

Looking at the source in [SCoverageReportMojo.java](https://github.com/scoverage/scoverage-maven-plugin/blob/dc7c6a6bc6285c8235516766352b1d18b9fa2546/src/main/java/org/scoverage/plugin/SCoverageReportMojo.java#L426C1-L426C1)
```java
Coverage coverage = Serializer.deserialize( coverageFile, project.getBasedir() );
```

For project `module01`, `project.getBasedir()` returns `C:\GIT\scoverage-maven-plugin\target\it\test_aggregate\module01`

This gets appended to the source path in the coverage file, producing the duplicated module name in the path.

This is as far as I got; I am not sure whether the 'bug' is that `module01` is present in the `scoverage.coverage` file when it should not be, or if the `project.getBasedir()` should be something like `getExecutionRoot(project).getBasedir()`.

I hope this helps you track down the issue some more.
